### PR TITLE
Update README badges

### PR DESCRIPTION
**Pull request by evan2645**
This PR adds the new SPIFFE badge indicating production readiness.

I also took the liberty of removing the coveralls badge, as the number
hasn't moved in years and I don't believe that the project has found
this metric to be very useful. Perhaps it can be made more useful, but
for now it seems like a good idea to just remove it. Happy to put it
        back if folks would prefer to keep it.

Signed-off-by: Evan Gilman <egilman@vmware.com>

### CODE REVIEW

**Comment by evan2645** [comment]
The syntax highlighters really love our new badge links

File: README.md [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 [![CII Best Practices](https://bestpractices.coreinfrastructure.org/projects/3303/badge)](https://bestpractices.coreinfrastructure.org/projects/3303)
 [![Build Status](https://github.com/spiffe/spire/actions/workflows/pr_build.yaml/badge.svg)](https://github.com/spiffe/spire/actions/workflows/pr_build.yaml)
-[![Coverage Status](https://coveralls.io/repos/github/spiffe/spire/badge.svg?branch=main)](https://coveralls.io/github/spiffe/spire?branch=main)
```
Comment (azdagron) [inline README.md:5]: Is dropping this intentional?

**Review by azdagron — APPROVED** [review]
Heh, yeah, that diff is sweet 😆 I had one question about the removal of code coverage, but tentatively approve.

File: README.md (hunk 1 above)
Comment (azdagron) [reply README.md:5]: Whoops. I only saw your second comment in my email chain. Please ignore. I agree with your assessment.

**Comment by azdagron** [comment]
CI/CD failure is a timeout on docker-compose and wholly unrelated to this change. Merging anyway.

**Pull request merged**